applications/nrf_desktop/src/modules/watchdog.c

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+#include <device.h>
+#include <drivers/watchdog.h>
+
+#define MODULE watchdog
+#include "module_state_event.h"
+#include "power_event.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_WATCHDOG_LOG_LEVEL);
+
+
+#if defined(CONFIG_WDT_0_NAME)
+#define WATCHDOG_DEV_NAME CONFIG_WDT_0_NAME
+#elif defined(DT_WDT_0_NAME)
+#define WATCHDOG_DEV_NAME DT_WDT_0_NAME
+#else
+#error "Error!!! Watchdog device name not defined - verify configuration."
+#endif
+#define WDT_FEED_WORKER_DELAY 2000 //2s
+#define WDT_FEED_WORKER_START_IMMEDIATELY 0
+
+#if defined (CONFIG_DESKTOP_WATCHDOG_TIMEOUT)
+#define WDT_TIMEOUT CONFIG_DESKTOP_WATCHDOG_TIMEOUT
+#else
+#define WDT_TIMEOUT 16000 //16s - required to survive LOG_PANIC print when power down
+#warning "missing config"
+#endif
+
+struct wdt_data_storage
```

Comment:
In other modules we do not use such struct, just `static` values.

Reply:
I decided to use structure to gather all globals in once place and make single point (one top most function) to get access to it. The function fowards adress of an instance of the structure to functions called that require access to global data. That is just one pointer in function calls so stack size should not be severly affected by that.

Reply:
ok